lib/base.js

```diff
@@ -309,9 +309,10 @@ class ConnectionPool extends EventEmitter {
     if (!this.pool) return setImmediate(callback, null)
 
     const pool = this.pool
-    this.pool.drain().then(() => {
-      pool.clear()
-      callback(null)
+    pool.drain().then(() => {
+      pool.clear().then(() => {
```

Comment:
```suggestion
      return pool.clear().then(() => {
```

Reply:
You should also catch and `callback(err)` if possible.

Reply:
> You should also catch and `callback(err)` if possible.

`_poolDestroy` doesn't throw